transports/quic/src/transport.rs

```diff
@@ -149,13 +185,16 @@ impl<P: Provider> Transport for GenTransport<P> {
         listener_id: ListenerId,
         addr: Multiaddr,
     ) -> Result<(), TransportError<Self::Error>> {
-        let (socket_addr, version, _peer_id) =
-            multiaddr_to_socketaddr(&addr, self.support_draft_29)
-                .ok_or(TransportError::MultiaddrNotSupported(addr))?;
+        let (socket_addr, version, _peer_id) = self.remote_multiaddr_to_socketaddr(addr, false)?;
+        let endpoint_config = self.quinn_config.endpoint_config.clone();
+        let server_config = self.quinn_config.server_config.clone();
+        let socket = UdpSocket::bind(socket_addr).map_err(Self::Error::from)?;
+        let socket_c = socket.try_clone().map_err(Self::Error::from)?;
```

Comment:
Thank you for cloning an existing socket instead of using port-reuse via an additional socket here @kpp .